datalad/support/annexrepo.py

```diff
@@ -12,23 +12,53 @@
 
 """
 
-from os.path import join, exists
+from os import getcwd
+from os.path import join as p_join, exists, normpath, isabs, commonprefix, relpath
 import logging
 
 from ConfigParser import NoOptionError
 
-from gitrepo import GitRepo
+from gitrepo import GitRepo, files_decorator
 from datalad.cmd import Runner as Runner
+from exceptions import CommandNotAvailableError, CommandError, FileNotInAnnexError, FileInGitError
 
 lgr = logging.getLogger('datalad.annex')
 
 
+def _options_decorator(func):
+    """Decorator to provide convenient way to pass options to command calls.
+
+    Any keyword argument "foo='bar'" translates to " --foo=bar".
+    All of these are collected in a list and then passed to keyword argument `options`
+    of the decorated function.
+    Note: This is meant to especially decorate the methods of AnnexRepo-class and therefore
+    returns a class method.
+    """
+
+    def newfunc(self, *args, **kwargs):
+        option_list=[]
+        for key in kwargs.keys():
+            option_list.extend([" --%s=%s" % (key, kwargs.get(key))])
+
+        return func(self, *args, options=option_list)
+    return newfunc
+
+
+
 class AnnexRepo(GitRepo):
     """Representation of an git-annex repository.
 
-    """
 
-    def __init__(self, path, url=None, runner=None):
+    Paths given to any of the class methods will be interpreted as relative to os.getcwd() in case this is currently beneath
```

Comment:
Please keep docstrings within PEP recommended 79 char width.  Currently seems to spill out quite a bit which complicates review on github